supervisor/backups/backup.py

```diff
@@ -178,12 +182,44 @@ def docker(self, value: dict[str, Any]) -> None:
         """Set the Docker config data."""
         self._data[ATTR_DOCKER] = value
 
-    @cached_property
+    @property
     def location(self) -> str | None:
         """Return the location of the backup."""
-        for backup_mount in self.sys_mounts.backup_mounts:
-            if self.tarfile.is_relative_to(backup_mount.local_where):
-                return backup_mount.name
+        return self.locations[0]
+
+    @property
+    def all_locations(self) -> set[str | None]:
+        """Return all locations this backup was found in."""
+        return self._locations.keys()
+
+    @property
+    def locations(self) -> list[str | None]:
+        """Return locations this backup was found in except cloud backup (unless that's the only one)."""
```

Comment:
Seems a bit weird to me that we only show the cloud location if it is the only one, isn't that a bit inconsistent?

Reply:
I just didn't want this list to be empty ever. If a backup exists exclusively in `.cloud_backup` no one will be aware of it except core. They're the only ones who will know its slug to ask more info about it and figured it was fine to put that location in the list in this case. In all other cases I would prefer not to show `.cloud_backup` as it will confuse all other clients who have no idea what that location is.

Reply:
> I just didn't want this list to be empty ever.  [...] In all other cases I would prefer not to show .cloud_backup as it will confuse all other clients who have no idea what that location is.

Hm, ok other clients might not know about cloud backup, but maybe they should, so nothing funky happens accidentally (e.g. if something alters the backup file? Then maybe it is better if it has the chance to know?)

I mean, I get your arguments. But I think rather than adding this complexity, and potentially hiding something from other users which might be(come) important, why not just leave it here. If it turns out to be a problem, we can remove it then. :man_shrugging: